packages/frontend/components/basic/BasicButton.vue

```diff
@@ -0,0 +1,14 @@
+<script setup lang="ts">
+interface Props {
+  text: string
+  type?: 'submit' | 'button'
+}
+
+const props = defineProps<Props>()
+</script>
+
+<template>
+  <button :type="type ?? 'button'" class="px-5 py-5 bg-primary rounded-md text-black font-semibold text-lg">
```

Comment:
we can use withDefaults for this
```suggestion
const props = withDefaults(defineProps<Props>(), {
  type: 'button'
})
</script>

<template>
  <button :type="type" class="px-5 py-5 bg-primary rounded-md text-black font-semibold text-lg">
```